jitsi-utils/src/main/java/org/jitsi/utils/queue/QueueStatistics.java

```diff
@@ -88,54 +108,65 @@ public JSONObject getStats()
     {
         JSONObject stats = new JSONObject();
         long now = System.currentTimeMillis();
-        stats.put("added", totalPacketsAdded.get());
-        stats.put("removed", totalPacketsRemoved.get());
-        stats.put("dropped", totalPacketsDropped.get());
+        stats.put("added", totalPacketsAdded.sum());
+        stats.put("removed", totalPacketsRemoved.sum());
+        stats.put("dropped", totalPacketsDropped.sum());
         stats.put("add_rate", addRate.getRate(now));
         stats.put("remove_rate", removeRate.getRate(now));
         stats.put("drop_rate", dropRate.getRate(now));
         double duration = (now - firstPacketAddedMs) / 1000d;
         stats.put("duration_s", duration);
-        stats.put("average_remove_rate_pps", totalPacketsRemoved.get() / duration);
+        stats.put("average_remove_rate_pps", totalPacketsRemoved.sum() / duration);
+        stats.put("average_queue_size_at_remove", totalSize.sum() / (double)totalPacketsRemoved.sum());
+        stats.put("average_queue_wait_time_ms", totalWait.sum() / (double)totalPacketsRemoved.sum());
 
         return stats;
     }
 
     /**
      * Registers the addition of a packet.
-     * @param now the time (in milliseconds since the epoch) at which the
-     * packet was added.
      */
-    public void add(long now)
+    @Override
+    public void added(Object o)
     {
+        long now = System.currentTimeMillis();
```

Comment:
Can we use a `Clock` instead?  I think at some point we'll want to replace all uses of `System.currentTimeMillis()` as it doesn't work well with tests.

Reply:
The thing I'm uncertain about with using `Clock` is that at some point I'd like to switch to using `System.nanoTime()` across our system for things that don't actually need wall-clock times, and as far as I know there's no `Clock` defined for that. (We could define one, of course, but it's a little tricky because some aspects of `Clock` assume that it has a 1970 epoch.)